.github/workflows/ispc-ci.yml

```diff
@@ -320,14 +323,105 @@ jobs:
+  # - lin-test-prior for older LLVM versions
   linux-test:
     needs: [define-flow, linux-build-ispc]
     runs-on: ubuntu-22.04
     continue-on-error: false
     strategy:
       fail-fast: false
       matrix:
-        version: [17, 18]
+        version: [18, 19]
+        arch: [x86, x86-64]
+        target: ${{fromJson(needs.define-flow.outputs.tests_matrix_targets_x86)}}
+        # See issue #2818 for more deatils. It's SDE problem running on AMD runner.
+        exclude:
+          - arch: x86
+            target: "avx2vnni-i32x4"
+          - arch: x86
+            target: "avx2vnni-i32x8"
+          - arch: x86
+            target: "avx2vnni-i32x16"
+          - arch: x86
+            target: "avx512skx-x4"
+          - arch: x86
+            target: "avx512skx-x8"
+          - arch: x86
+            target: "avx512skx-x16"
+          - arch: x86
+            target: "avx512skx-x32"
+          - arch: x86
+            target: "avx512skx-x64"
+          - arch: x86
+            target: "avx512icl-x4"
+          - arch: x86
+            target: "avx512icl-x8"
+          - arch: x86
+            target: "avx512icl-x16"
+          - arch: x86
+            target: "avx512icl-x32"
+          - arch: x86
+            target: "avx512icl-x64"
+          - arch: x86
+            target: "avx512spr-x4"
+          - arch: x86
+            target: "avx512spr-x8"
+          - arch: x86
+            target: "avx512spr-x16"
+          - arch: x86
+            target: "avx512spr-x32"
+          - arch: x86
+            target: "avx512spr-x64"
+
+    steps:
+    - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+    - name: Download package
+      uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+      with:
+        name: ispc_llvm${{matrix.version}}_linux
+
+    - name: Install dependencies and unpack artifacts
+      run: |
+        .github/workflows/scripts/install-test-deps.sh
+
+    - name: Check environment
+      run: |
+        cat /proc/cpuinfo
+
+    - name: Running tests
+      run: |
+        echo PATH="$PATH"
+        ./scripts/alloy.py -r --only="stability ${{ matrix.arch }} current ${{ needs.define-flow.outputs.tests_optsets }}" --only-targets="${{ matrix.target }}" --time --update-errors=FP
+
+    - name: Check
+      run: |
+        # Print fails to the log.
+        git diff --exit-code
+
+    - name: Upload fail_db.txt
+      uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
+      if: failure()
+      with:
+        name: fail_db.llvm${{matrix.version}}.${{matrix.arch}}.${{matrix.target}}.txt
+        path: tests/fail_db.txt
+
+    - name: Upload alloy logs
+      uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
+      if: failure()
+      with:
+        name: alloy_results.llvm${{matrix.version}}.${{matrix.arch}}.${{matrix.target}}
+        path: alloy_results_*
+
+  linux-test-prior:
```

Comment:
I don't really like this approach. This workflow file is already huge, and this only makes it bigger. What about making it reusable with the LLVM version as a parameter and creating separate workflows for each LLVM version that call this one? Basically, the same approach we use for rebuild-llvm jobs.

Reply:
I created reusable workflow for test jobs. I'd like to do the same for build but I'll do it in another PR for easier review.
The full run is here: https://github.com/ispc/ispc/actions/runs/13713528891

Reply:
It is really nice. It is much more convenient to navigate the job list now.